relay-sampling/src/lib.rs

```diff
@@ -993,8 +995,15 @@ pub fn merge_configs_and_match(
     // Keep in mind that the sample rate received here has already been derived by the matching
     // logic, based on multiple matches and decaying functions.
     //
-    // We also decide to use the sampling mode of the project to which the event belongs.
-    match_result.set_sample_rate(match sampling_config.mode {
+    // The determination of the sampling mode occurs with the following priority:
+    // 1. Non-root project sampling mode
+    // 2. Root project sampling mode
+    // 3. SamplingMode::Unsupported
+    let sampling_mode = sampling_config.map_or(
+        root_sampling_config.map_or(SamplingMode::Unsupported, |config| config.mode),
+        |config| config.mode,
+    );
```

Comment:
I would make this failure more explicit:

```suggestion
    let Some(primary_config) = sampling_config.or(root_sampling_config) else {
        relay_log::error!("Cannot sample without config");
        return None;
    }
    let sampling_mode = primary_config.sampling_mode;
```

Reply:
Damn, 10x more expressive, I love it!